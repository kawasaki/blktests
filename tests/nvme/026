#!/bin/bash
# SPDX-License-Identifier: GPL-2.0+
# Copyright (c) 2017-2018 Western Digital Corporation or its affiliates.
#
# Test NVMe ns-descs command.

. tests/nvme/rc

DESCRIPTION="test NVMe ns-descs"
QUICK=1

requires() {
	_nvme_requires
	_have_loop
	_require_nvme_trtype_is_fabrics
}

set_conditions() {
	_set_nvme_trtype_and_nvmet_blkdev_type "$@"
}

test() {
	echo "Running ${TEST_NAME}"

	_setup_nvmet

	local ns

	_nvmet_target_setup

	_nvme_connect_subsys

	ns=$(_find_nvme_ns "${def_subsys_uuid}")

	if ! nvme ns-descs "/dev/${ns}" >> "$FULL" 2>&1; then
		echo "ERROR: ns-desc failed"
	fi

	_nvme_disconnect_subsys >> "$FULL" 2>&1

	_nvmet_target_cleanup

	echo "Test complete"
}
