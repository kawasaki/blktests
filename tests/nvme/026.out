Running nvme/026
Test complete
